ui/app/serializers/volume.js

```diff
@@ -23,17 +23,16 @@ export default class VolumeSerializer extends ApplicationSerializer {
     hash.ID = JSON.stringify([`csi/${hash.ID}`, hash.NamespaceID || 'default']);
     hash.PluginID = `csi/${hash.PluginID}`;
 
-    // Convert hash-based allocation embeds to lists
+    // Populate read/write allocation lists from aggregate allocation list
     const readAllocs = hash.ReadAllocs || {};
     const writeAllocs = hash.WriteAllocs || {};
-    const bindIDToAlloc = hash => id => {
-      const alloc = hash[id];
-      alloc.ID = id;
+    const bindIDToAlloc = allocList => id => {
+      const alloc = allocList.find(a => a.ID === id);
```

Comment:
I'm not sure if the allocs list will ever be large enough for this to be significant, but using a `find` here changes the complexity from linear to quadratic in the worst case. 

It should be possible to invert this and iterate over allocs and build the `ReadAllocations` and `WriteAllocations` representations by checking for existence in the `ReadAllocs` and `WriteAllocs` hashes. That would make this linear again.

Reply:
Fixed in 198264e